packages/api/src/router/absence.ts

```diff
@@ -0,0 +1,59 @@
+import { z } from "zod";
+
+import { createTRPCRouter, protectedProcedure } from "../trpc";
+
+export const absenceRouter = createTRPCRouter({
+  createAbsence: protectedProcedure
+    .input(
+      z.object({
+        userId: z.number().nonnegative(),
+        absenceDate: z.date(),
+      }),
+    )
+    .mutation(async ({ input, ctx }) => {
+      await ctx.db.absence.create({
+        data: {
+          userId: input.userId,
+          absenceDate: input.absenceDate,
+        },
+      });
+    }),
+
+  getAllAbsences: protectedProcedure.query(async ({ ctx }) => {
+    return await ctx.db.absence.findMany();
+  }),
```

Comment:
can you convert this to an adminProcedure, then create another getAbsences that uses a userId?